components/components.go

```diff
@@ -61,6 +62,19 @@ func (c *Component) IsRunning() (bool, error) {
 	return docker.IsRunning(c.Name, c.ImageWithVersion())
 }
 
+// GetPorts returns component ports of the component if there is any
+func (c *Component) GetPorts() ([]types.Port, error) {
```

Comment:
For consistency, instead of dealing with docker api types in the cli, maybe we could do `GetPublicPorts() ([]uint16, error)` and simplify `publicPortsFmt`.

Reply:
it was my original idea and actually I even implement it like that first. But then I figured we might need a private port in the future.

for example to connect from `mysql-client` to gitbase inside the docker you need to know private port. Currently, it just matches default one but we might need something similar in the future for other components.

To solve this issue we can actually hard-code internal port because we pin all the images. But if we are going to implement something like https://github.com/src-d/backlog/issues/1364 we will need this method.

Reply:
I see. Should we do a type alias for `types.Port`? In our docker package we do 
```go
type Container = types.Container
type Volume = types.Volume
```

But in this case I don't know if we should do `docker.Port` (consistent with other docker types aliases, but the method  is in `components` package), a `components.Port`, or just leave it as it is...

Reply:
No opinion on it. I'm good with adding the alias as well as keeping using `docker/types`.

//cc @se7entyse7en wdyt?

Reply:
I also prefer the alias and I'd put it in the `docker` package. IMO is more suitable there instead of adding it in the `components` package.